src/stores/maintenanceTaskStore.ts

```diff
@@ -0,0 +1,148 @@
+import type { InstallValidation } from '@comfyorg/comfyui-electron-types'
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+
+import { DESKTOP_MAINTENANCE_TASKS } from '@/constants/desktopMaintenanceTasks'
+import type {
+  MaintenanceTask,
+  MaintenanceTaskState
+} from '@/types/desktop/maintenanceTypes'
+import { electronAPI } from '@/utils/envUtil'
+import { useMinLoadingDurationRef } from '@/utils/refUtil'
+
+/**
+ * User-initiated maintenance tasks.  Currently only used by the desktop app maintenance view.
+ *
+ * Includes running state, task list, and execution / refresh logic.
+ * @returns The maintenance task store
+ */
+export const useMaintenanceTaskStore = defineStore('maintenanceTask', () => {
+  /** Refresh should run for at least this long, even if it completes much faster. Ensures refresh feels like it is doing something. */
+  const minRefreshTime = 250
+  const electron = electronAPI()
+
+  // Reactive state
+  const isRefreshing = useMinLoadingDurationRef(false, minRefreshTime)
```

Comment:
Let's move the call of `useMinLoadingDurationRef` to Vue component. The pinia store should save the gound truth of states.